### PR TITLE
Add AVX512 code for high bd forward transform

### PR DESCRIPTION
**Pull request by kirithika7**
This PR has AVX512 high bd forward transform kernels for sizes 16x16,32x32,64x64,64x32,32x64,64x16,16x64,32x16,16x32 and their unit tests.The performance numbers are given in the commit messages,

### CODE REVIEW

**Comment by kirithika7** [comment]
It builds and executes fine with gcc 7.3.0 but the Travis test with gcc 5.4.0 fails.

**Comment by 1480c1** [comment]
It is possible to update the gcc and clang using the apt sources provided by ubuntu and llvm or using a docker,

**Comment by kirithika7** [comment]
> It is possible to update the gcc and clang using the apt sources provided by ubuntu and llvm or using a docker,

I have updated the gcc and tested locally.It is building and executing without any errors.

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 1]
```diff
@@ -40,7 +40,7 @@ extern "C" {
 #endif
 
 // Internal Marcos
-#define NON_AVX512_SUPPORT
+//#define NON_AVX512_SUPPORT
```
Comment (hassount) [inline Source/Lib/Common/Codec/EbDefinitions.h]: this should be put back on for now

**Comment by JackZhouVCD** [comment]
After applying this PR, the encoder crashes during the dealloc process. This issue was addressed in #499 which is later on in the master than the base commit shown in this PR. Please rebase to sync up with master.

**Pull request merged**